core/value.mli

```diff
@@ -55,16 +55,20 @@ and xml = xmlitem list
 type table = (database * string) * string * string list list * Types.row
   [@@deriving show]
 
+type primitive_value_basis =  [
+  | `Bool of bool
+  | `Char of char
+  | `Float of float
+  | `Int of int
+  | `XML of xmlitem
+  | `String of string ]
+  [@@deriving show]
+
 type primitive_value = [
-| `Bool of bool
-| `Char of char
-| `Database of (database * string)
-| `Table of table
-| `Float of float
-| `Int of int
-| `XML of xmlitem
-| `String of string ]
-[@@deriving show]
+  | primitive_value_basis
+  | `Database of (database * string)
+  | `Table of table ]
+  [@@deriving show]
```

Comment:
@jstolarek expressed concern regarding this change. I will argue it changes very little or nothing, because this is exactly the implementation in `value.ml`. This minor change reduces the amount of necessary boilerplate code in the `Compressible_Value` module in `serialisation.ml`.